src/StreamingDataFrames/streamingdataframes/models/serializers/quix.py

```diff
@@ -54,12 +51,44 @@ class QCodecId:
     SUPPORTED_CODECS = (JSON, JSON_TYPED)
 
 
+class LegacyHeaders:
+    Q_CODEC_ID = "C"
+    Q_MODEL_KEY = "K"
+    VALUE = "V"
+    VALUE_START_IDX = "S"
+    VALUE_END_IDX = "E"
+
+
 def _b64_decode_or_none(s) -> Optional[bytes]:
     if s is not None:
         return base64.b64decode(s)
 
 
-class BaseQuixDeserializer(JSONDeserializer, metaclass=ABCMeta):
+class QuixDeserializer(JSONDeserializer):
+    def __init__(
+        self,
+        column_name: Optional[str] = None,
+        loads: Optional[Callable[[Any, None], Union[str, bytes]]] = None,
+        loads_kwargs: Optional[Mapping] = None,
+    ):
+        """
+        Parses JSON data from either Timeseries/Parameter or EventData formats.
+
+        :param column_name: if provided, the deserialized value will be wrapped into
+            dictionary with `column_name` as a key.
+        :param loads: function to parse json from bytes. Default - `json.loads`.
+        :param loads_kwargs: dict with named arguments for `loads` function.
+        """
+        super().__init__(
+            column_name=column_name, loads=loads, loads_kwargs=loads_kwargs
+        )
+        self._deserializers = {
+            QModelKey.TIMESERIESDATA: self.deserialize_timestamp,
+            QModelKey.PARAMETERDATA: self.deserialize_timestamp,
```

Comment:
Maybe call it `deserialize_timeseries`?

Reply:
lol, IDK how I accidently called it that. Must have had timestamps on the brain =)